### DISCUSSION TITLE
GitHub Actions badge is always "no status" in some repositories

### DISCUSSION THREAD

**Discussion by nokok** [community/community#14980] [Actions] [answered]
I am experiencing this problem in several private repositories of my organization. When I interact with the "Create status badge" UI, it always shows "no status". The URL generated is

`https://github.com/<org>/<repo>/actions/workflows/<filename>/badge.svg`

In fact, this Action is successful and displaying "passing" is the expected behavior.

However, if you change the URL to the one below, it will correctly display "passing".
`https://github.com/<org>/<repo>/workflows/<workflow-name>/badge.svg`

Perhaps there is a problem with the link generated by the UI.

**Comment by remarkablemark** [comment]
I'm experiencing the same issue with my public repository: https://github.com/remarkablemark/paccurate

All my workflow status badges show `no status`. Here's an example from `Create status badge`:

[![build](https://github.com/remarkablemark/paccurate/actions/workflows/build.yml/badge.svg)](https://github.com/remarkablemark/paccurate/actions/workflows/build.yml)

I can confirm that when using the URL:

```
https://github.com/<org>/<repo>/actions/workflows/<filename>/badge.svg
```

The status badge shows the correct status:

[![build](https://github.com/remarkablemark/paccurate/workflows/build/badge.svg)](https://github.com/remarkablemark/paccurate/actions/workflows/build.yml)

**Comment by nokok** [comment, marked as answer]
Is it fixed? 👀

**Reply by remarkablemark** [reply]
Seems like it is for me!

**Reply by aespitia** [reply]
it is not :(

**Reply by Gustl22** [reply]
They changed the path to:
`https://github.com/OWNER/REPOSITORY/`**actions**`/workflows/WORKFLOW-FILE`**.yml**`/badge.svg?branch=BRANCH`

**Comment by marcindulak** [comment]
The problem still exists. The "Create status badge" button generates a link showing "no status"

```
https://github.com/<org>/<repo>/actions/workflows/<filename>/badge.svg
```

and manually editing the link into the one below brings back the status

```
https://github.com/<org>/<repo>/workflows/<workflow-name>/badge.svg
```

**Reply by Slion** [reply]
Still have that issue and the workaround works. Make sure you get the case of your workflow name right.

**Comment by Zibri** [comment]
Same for me... the default is "passing" the release is "no status"

**Comment by aespitia** [comment]
i still have the same problem, it used to work, but now it hasn't for some time.

**Reply by aespitia** [reply]
what I found is that the query parameter ?branch=[branch name] no longer works...it used to though.  After I took off the query parameter, I can get status again, but only for 1 branch, so i'm assuming it's the default dev branch

**Reply by GAFelton** [reply]
I am having the same issue - `?branch=[branch_name]` no longer works - it just displays "no status" regardless

**Reply by mosman128** [reply]
Removing branch name worked for me 🎉  Thank you @aespitia

**Reply by NeonWizard** [reply]
How can we get the status badge for a branch though?

**Reply by aespitia** [reply]
that's what we're saying is the bug, but has yet to be addressed.

**Reply by smped** [reply]
Still not fixed by the looks. An out-of-date badge exists for the main branch, but after shifting my tests to a specific `gh-actions` branch, there's no badge on `?branch=gh-actions`

**Comment by RPGillespie6** [comment]
Just another "me too", no matter what I try it's always "no status" even though the builds are passing:

https://github.com/RPGillespie6/fastcov/actions/workflows/build-test.yaml/badge.svg

vs.

https://github.com/RPGillespie6/fastcov/workflows/Build%20and%20Test/badge.svg

Workaround doesn't work for me.

**Reply by RPGillespie6** [reply]
I was able to get it to work by renaming the workflow to not have spaces. Now the workaround works:

https://github.com/RPGillespie6/fastcov/workflows/build/badge.svg

**Comment by Gustl22** [comment]
They changed the path from:

`https://github.com/OWNER/REPOSITORY/workflows/WORKFLOW-NAME/badge.svg?branch=BRANCH`

to:

`https://github.com/OWNER/REPOSITORY/`**actions**`/workflows/WORKFLOW-FILE`**.yml**`/badge.svg?branch=BRANCH`

See [docs](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/monitoring-workflows/adding-a-workflow-status-badge).
